pkg/roll/execute.go

```diff
@@ -112,6 +112,9 @@ func (m *Roll) StartDDLOperations(ctx context.Context, migration *migrations.Mig
 func (m *Roll) ensureViews(ctx context.Context, schema *schema.Schema, version string) error {
 	// create schema for the new version
 	versionSchema := VersionedSchemaName(m.schema, version)
+	if err := migrations.ValidateName(versionSchema); err != nil {
+		return err
+	}
```

Comment:
This is too late to check the length of the new version schema - at this point a potentially lengthy migration start operation has already taken place so it's a poor user experience for the migration to fail with a validation check at this stage. I'd remove the check here and if we need it we can add a check earlier in the migration start process as a separate PR.